diff --git a/node_modules/@bacons/link-assets/build/index.js b/node_modules/@bacons/link-assets/build/index.js
index 2547de6..ad3d633 100644
--- a/node_modules/@bacons/link-assets/build/index.js
+++ b/node_modules/@bacons/link-assets/build/index.js
@@ -11,12 +11,13 @@ const groupFiles_1 = require("./groupFiles");
 const withLinkedAsset = (config, props) => {
     const expanded = props
         .map((filePath) => {
-        if (fs_1.default.statSync(filePath).isDirectory()) {
+        const resolved = path_1.default.resolve(config._internal?.projectRoot, filePath);
+        if (fs_1.default.statSync(resolved).isDirectory()) {
             return fs_1.default
-                .readdirSync(filePath)
-                .map((file) => path_1.default.join(filePath, file));
+                .readdirSync(resolved)
+                .map((file) => path_1.default.join(resolved, file));
         }
-        return [filePath];
+        return [resolved];
     })
         .flat();
     debug("All files:", expanded);
