diff --git a/node_modules/@react-native-menu/menu/ios/MenuView.swift b/node_modules/@react-native-menu/menu/ios/MenuView.swift
index 4400f47..a130fab 100644
--- a/node_modules/@react-native-menu/menu/ios/MenuView.swift
+++ b/node_modules/@react-native-menu/menu/ios/MenuView.swift
@@ -17,12 +17,23 @@ class MenuView: UIButton {
                 return
             }
             _actions.removeAll()
-            actions.forEach { menuAction in
+            actions.reversed().forEach { menuAction in
                 _actions.append(RCTMenuAction(details: menuAction).createUIMenuElement({action in self.sendButtonAction(action)}))
             }
             self.setup()
         }
     }
+    @objc var colorMode: NSString? {
+        didSet {
+            if (colorMode == "dark") {
+                self.overrideUserInterfaceStyle = .dark
+            } else if (colorMode == "light") {
+                self.overrideUserInterfaceStyle = .light
+            }
+            
+            self.setup()
+        }
+    }
 
     private var _title: String = "";
     @objc var title: NSString? {
@@ -34,6 +45,7 @@ class MenuView: UIButton {
             self.setup()
         }
     }
+    
     @objc var onPressAction: RCTDirectEventBlock?
     
     @objc var shouldOpenOnLongPress: Bool = false {
diff --git a/node_modules/@react-native-menu/menu/ios/RCTUIMenuManager.m b/node_modules/@react-native-menu/menu/ios/RCTUIMenuManager.m
index 45543d8..a61716c 100644
--- a/node_modules/@react-native-menu/menu/ios/RCTUIMenuManager.m
+++ b/node_modules/@react-native-menu/menu/ios/RCTUIMenuManager.m
@@ -20,4 +20,6 @@ RCT_EXPORT_VIEW_PROPERTY(onPressAction, RCTDirectEventBlock);
  */
 RCT_EXPORT_VIEW_PROPERTY(shouldOpenOnLongPress, BOOL)
 
+RCT_EXPORT_VIEW_PROPERTY(colorMode, NSString)
+
 @end
