diff --git a/node_modules/@react-navigation/elements/src/Header/Header.tsx b/node_modules/@react-navigation/elements/src/Header/Header.tsx
index 12c570d..d362dee 100644
--- a/node_modules/@react-navigation/elements/src/Header/Header.tsx
+++ b/node_modules/@react-navigation/elements/src/Header/Header.tsx
@@ -24,6 +24,10 @@ type Props = HeaderOptions & {
    * Title text for the header.
    */
   title: string;
+  /**
+   * Can go back.
+   */
+  canGoBack?: boolean;
 };
 
 const warnIfHeaderStylesDefined = (styles: Record<string, any>) => {
@@ -52,6 +56,7 @@ export default function Header(props: Props) {
     layout = frame,
     modal = false,
     title,
+    canGoBack,
     headerTitle: customTitle,
     headerTitleAlign = Platform.select({
       ios: 'center',
@@ -191,6 +196,7 @@ export default function Header(props: Props) {
         pressColor: headerPressColor,
         pressOpacity: headerPressOpacity,
         labelVisible: headerLeftLabelVisible,
+        canGoBack
       })
     : null;
 
@@ -199,6 +205,7 @@ export default function Header(props: Props) {
         tintColor: headerTintColor,
         pressColor: headerPressColor,
         pressOpacity: headerPressOpacity,
+        canGoBack
       })
     : null;
 
diff --git a/node_modules/@react-navigation/elements/src/types.tsx b/node_modules/@react-navigation/elements/src/types.tsx
index 91e8511..8c665f4 100644
--- a/node_modules/@react-navigation/elements/src/types.tsx
+++ b/node_modules/@react-navigation/elements/src/types.tsx
@@ -37,12 +37,7 @@ export type HeaderOptions = {
   /**
    * Function which returns a React Element to display on the left side of the header.
    */
-  headerLeft?: (props: {
-    tintColor?: string;
-    pressColor?: string;
-    pressOpacity?: number;
-    labelVisible?: boolean;
-  }) => React.ReactNode;
+  headerLeft?: (props: HeaderButtonProps) => React.ReactNode;
   /**
    * Whether a label is visible in the left button. Used to add extra padding.
    */
@@ -54,11 +49,7 @@ export type HeaderOptions = {
   /**
    * Function which returns a React Element to display on the right side of the header.
    */
-  headerRight?: (props: {
-    tintColor?: string;
-    pressColor?: string;
-    pressOpacity?: number;
-  }) => React.ReactNode;
+  headerRight?: (props: HeaderButtonProps) => React.ReactNode;
   /**
    * Style object for the container of the `headerRight` element.
    */
