diff --git a/node_modules/react-native-tab-view/src/PanResponderAdapter.tsx b/node_modules/react-native-tab-view/src/PanResponderAdapter.tsx
index 3d1964a..6a12ca1 100644
--- a/node_modules/react-native-tab-view/src/PanResponderAdapter.tsx
+++ b/node_modules/react-native-tab-view/src/PanResponderAdapter.tsx
@@ -268,14 +268,17 @@ export default function PanResponderAdapter<T extends Route>({
   });
 
   const maxTranslate = layout.width * (routes.length - 1);
-  const translateX = Animated.multiply(
-    panX.interpolate({
-      inputRange: [-maxTranslate, 0],
-      outputRange: [-maxTranslate, 0],
-      extrapolate: 'clamp',
-    }),
-    I18nManager.isRTL ? -1 : 1
-  );
+  const translateX = React.useMemo(() => {
+    if (maxTranslate <= 0) return 0;
+    return Animated.multiply(
+      panX.interpolate({
+        inputRange: [-maxTranslate, 0],
+        outputRange: [-maxTranslate, 0],
+        extrapolate: 'clamp',
+      }),
+      I18nManager.isRTL ? -1 : 1
+    );
+  }, [maxTranslate, panX]);
 
   return children({
     position: layout.width
diff --git a/node_modules/react-native-tab-view/src/TabBar.tsx b/node_modules/react-native-tab-view/src/TabBar.tsx
index f1137f4..55bb7af 100644
--- a/node_modules/react-native-tab-view/src/TabBar.tsx
+++ b/node_modules/react-native-tab-view/src/TabBar.tsx
@@ -141,6 +141,9 @@ export default class TabBar<T extends Route> extends React.Component<
     tabWidths: { [key: string]: number },
     flattenedWidth: string | number | undefined
   ) => {
+    if(routes.length === 0){
+      return 0
+    }
     if (flattenedWidth === 'auto') {
       return tabWidths[routes[index].key] || 0;
     }
