diff --git a/node_modules/react-native-reanimated/lib/module/reanimated2/js-reanimated/index.js b/node_modules/react-native-reanimated/lib/module/reanimated2/js-reanimated/index.js
index da91ad9..2621063 100644
--- a/node_modules/react-native-reanimated/lib/module/reanimated2/js-reanimated/index.js
+++ b/node_modules/react-native-reanimated/lib/module/reanimated2/js-reanimated/index.js
@@ -1,4 +1,6 @@
 import JSReanimated from './JSReanimated';
+import createReactDOMStyle from 'react-native-web/dist/exports/StyleSheet/compiler/createReactDOMStyle';
+import { createTransformValue } from 'react-native-web/dist/exports/StyleSheet/preprocess';
 const reanimatedJS = new JSReanimated();
 global._makeShareableClone = c => c;
 global._scheduleOnJS = queueMicrotask;
@@ -11,8 +13,8 @@ export const _updatePropsJS = (updates, viewRef) => {
       acc[index][key] = value;
       return acc;
     }, [{}, {}]);
-    if (typeof component.setNativeProps === 'function') {
-      setNativeProps(component, rawStyles);
+    if (component.style !== undefined) {
+      updatePropsDOM(component, rawStyles);
     } else if (Object.keys(component.props).length > 0) {
       Object.keys(component.props).forEach(key => {
         if (!rawStyles[key]) {
@@ -26,16 +28,21 @@ export const _updatePropsJS = (updates, viewRef) => {
     }
   }
 };
-const setNativeProps = (component, style) => {
+const updatePropsDOM = (component, style) => {
   const previousStyle = component.previousStyle ? component.previousStyle : {};
   const currentStyle = {
     ...previousStyle,
     ...style
   };
   component.previousStyle = currentStyle;
-  component.setNativeProps({
-    style: currentStyle
-  });
+  const domStyle = createReactDOMStyle(currentStyle);
+  for (const key in domStyle) {
+    if(key === 'transform') {
+        (component.style).transform = createTransformValue(domStyle[key]);
+    }else{
+        (component.style)[key] = domStyle[key];
+    }
+  }
 };
 export default reanimatedJS;
 //# sourceMappingURL=index.js.map
\ No newline at end of file
diff --git a/node_modules/react-native-reanimated/src/reanimated2/mappers.ts b/node_modules/react-native-reanimated/src/reanimated2/mappers.ts
index 4e66106..bc8712f 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/mappers.ts
+++ b/node_modules/react-native-reanimated/src/reanimated2/mappers.ts
@@ -80,8 +80,11 @@ export function createMapperRegistry() {
   }
 
   function mapperRun() {
-    processingMappers = true;
     runRequested = false;
+    if (processingMappers) {
+      return;
+    }
+    processingMappers = true;
     if (mappers.size !== sortedMappers.length) {
       updateMappersOrder();
     }
