diff --git a/node_modules/react-native/Libraries/Blob/BlobManager.js b/node_modules/react-native/Libraries/Blob/BlobManager.js
index e125123..4042cfd 100644
--- a/node_modules/react-native/Libraries/Blob/BlobManager.js
+++ b/node_modules/react-native/Libraries/Blob/BlobManager.js
@@ -16,6 +16,7 @@ const BlobRegistry = require('./BlobRegistry');
 import type {BlobData, BlobOptions, BlobCollector} from './BlobTypes';
 import NativeBlobModule from './NativeBlobModule';
 import invariant from 'invariant';
+import { getBlobForArrayBuffer } from 'react-native-blob-jsi-helper';
 
 /*eslint-disable no-bitwise */
 /*eslint-disable eqeqeq */
@@ -71,9 +72,7 @@ class BlobManager {
         part instanceof ArrayBuffer ||
         (global.ArrayBufferView && part instanceof global.ArrayBufferView)
       ) {
-        throw new Error(
-          "Creating blobs from 'ArrayBuffer' and 'ArrayBufferView' are not supported",
-        );
+        return getBlobForArrayBuffer(part);
       }
       if (part instanceof Blob) {
         return {
diff --git a/node_modules/react-native/Libraries/Blob/FileReader.js b/node_modules/react-native/Libraries/Blob/FileReader.js
index 15a8b44..b46ef8d 100644
--- a/node_modules/react-native/Libraries/Blob/FileReader.js
+++ b/node_modules/react-native/Libraries/Blob/FileReader.js
@@ -12,6 +12,7 @@
 
 const Blob = require('./Blob');
 const EventTarget = require('event-target-shim');
+const { toByteArray } = require('base64-js');
 
 import NativeFileReaderModule from './NativeFileReaderModule';
 
@@ -82,7 +83,32 @@ class FileReader extends (EventTarget(...READER_EVENTS): any) {
   }
 
   readAsArrayBuffer() {
-    throw new Error('FileReader.readAsArrayBuffer is not implemented');
+    this._aborted = false;
+
+    if (blob == null) {
+      throw new TypeError(
+        "Failed to execute 'readAsArrayBuffer' on 'FileReader': parameter 1 is not of type 'Blob'",
+      );
+    }
+
+    NativeFileReaderModule.readAsDataURL(blob.data).then(
+      (dataUrl: string) => {
+        if (this._aborted) {
+          return;
+        }
+        const base64 = dataUrl.split(',')[1];
+        const buffer = toByteArray(base64);
+        this._result = buffer;
+        this._setReadyState(DONE);
+      },
+      error => {
+        if (this._aborted) {
+          return;
+        }
+        this._error = error;
+        this._setReadyState(DONE);
+      }
+    );
   }
 
   readAsDataURL(blob: ?Blob) {