diff --git a/node_modules/react-native-gesture-handler/lib/commonjs/components/GestureComponents.web.js b/node_modules/react-native-gesture-handler/lib/commonjs/components/GestureComponents.web.js
index 0a879ab..108cdf4 100644
--- a/node_modules/react-native-gesture-handler/lib/commonjs/components/GestureComponents.web.js
+++ b/node_modules/react-native-gesture-handler/lib/commonjs/components/GestureComponents.web.js
@@ -31,12 +31,12 @@ const Switch = (0, _createNativeWrapper.default)(_reactNative.Switch, {
 exports.Switch = Switch;
 const TextInput = (0, _createNativeWrapper.default)(_reactNative.TextInput);
 exports.TextInput = TextInput; 
-const DrawerLayoutAndroid = (0, _createNativeWrapper.default)(_reactNative.DrawerLayoutAndroid, {
-  disallowInterruption: true
-}); // @ts-ignore -- TODO(TS) to investigate if it's needed
+const DrawerLayoutAndroid = () => {
+  console.warn('DrawerLayoutAndroid is not supported on web!');
+  return React.createElement(View)
+}
 
 exports.DrawerLayoutAndroid = DrawerLayoutAndroid;
-DrawerLayoutAndroid.positions = _reactNative.DrawerLayoutAndroid.positions; // RefreshControl is implemented as a functional component, rendering a View
 // NativeViewGestureHandler needs to set a ref on its child, which cannot be done
 // on functional components
 
diff --git a/node_modules/react-native-gesture-handler/lib/module/components/GestureComponents.web.js b/node_modules/react-native-gesture-handler/lib/module/components/GestureComponents.web.js
index d773992..a5f19b4 100644
--- a/node_modules/react-native-gesture-handler/lib/module/components/GestureComponents.web.js
+++ b/node_modules/react-native-gesture-handler/lib/module/components/GestureComponents.web.js
@@ -12,11 +12,11 @@ export const Switch = createNativeWrapper(RNSwitch, {
   disallowInterruption: true
 });
 export const TextInput = createNativeWrapper(RNTextInput);
-export const DrawerLayoutAndroid = createNativeWrapper(RNDrawerLayoutAndroid, {
-  disallowInterruption: true
-}); // @ts-ignore -- TODO(TS) to investigate if it's needed
+export const DrawerLayoutAndroid = ()=>{
+  console.warn('DrawerLayoutAndroid is not supported on web!');
+  return React.createElement(View)
+}
 
-DrawerLayoutAndroid.positions = RNDrawerLayoutAndroid.positions; // RefreshControl is implemented as a functional component, rendering a View
 // NativeViewGestureHandler needs to set a ref on its child, which cannot be done
 // on functional components
 
