diff --git a/node_modules/react-native-share/ios/GenericShare.m b/node_modules/react-native-share/ios/GenericShare.m
index 45b49df..3f1c8af 100644
--- a/node_modules/react-native-share/ios/GenericShare.m
+++ b/node_modules/react-native-share/ios/GenericShare.m
@@ -17,7 +17,7 @@ - (void)shareSingle:(NSDictionary *)options
     inAppBaseUrl:(NSString *)inAppBaseUrl {
 
     NSLog(@"Try open view");
-    if(![serviceType isEqualToString:@"com.apple.social.twitter"] && [[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:inAppBaseUrl]]) {
+    if([serviceType isEqualToString:@"com.apple.social.twitter"]) {
         SLComposeViewController *composeController = [SLComposeViewController  composeViewControllerForServiceType:serviceType];
 
         NSURL *URL = [RCTConvert NSURL:options[@"url"]];
